File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/language/languages/CodeWhispererJavaScript.kt
// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package software.aws.toolkits.jetbrains.services.codewhisperer.language.languages

import software.aws.toolkits.jetbrains.services.codewhisperer.language.CodeWhispererProgrammingLanguage
import software.aws.toolkits.telemetry.CodewhispererLanguage

class CodeWhispererJavaScript private constructor() : CodeWhispererProgrammingLanguage() {
    override val languageId: String = ID

    override fun toTelemetryType(): CodewhispererLanguage = CodewhispererLanguage.Javascript

    override fun isCodeCompletionSupported(): Boolean = true

    override fun isImportAdderSupported(): Boolean = true

    override fun isClassifierSupported(): Boolean = true

    companion object {
        const val ID = "javascript"

        val INSTANCE = CodeWhispererJavaScript()
    }
}

File: src/codewhisperer/language/codewhispererJavascript.ts
/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CodewhispererLanguage } from '../../shared/telemetry/telemetry.gen'
import { CodeWhispererProgrammingLanguage } from './codewhispererProgrammingLanguage'

export class CodeWhispererJavascript extends CodeWhispererProgrammingLanguage {
    id: CodewhispererLanguage = 'javascript'

    toCodeWhispererRuntimeLanguage(): CodeWhispererProgrammingLanguage {
        return this
    }

    isCodeCompletionSupported(): boolean {
        return true
    }

    isCodeScanSupported(): boolean {
        return false
    }
}

export class CodeWhispererJsx extends CodeWhispererProgrammingLanguage {
    id: CodewhispererLanguage = 'jsx'

    toCodeWhispererRuntimeLanguage(): CodeWhispererProgrammingLanguage {
        return new CodeWhispererJavascript()
    }

    isCodeCompletionSupported(): boolean {
        return true
    }

    isCodeScanSupported(): boolean {
        return false
    }
}
